src/compiler/output-targets/dist-custom-elements/index.ts

```diff
@@ -329,11 +334,47 @@ const getCustomElementCustomTransformer = (
     style: 'static',
     styleImportData: 'queryparams',
   };
-  return [
+  const beforeCustomTransformers = [
     addDefineCustomElementFunctions(compilerCtx, components, outputTarget),
     updateStencilCoreImports(transformOpts.coreImportPath),
+  ];
+
+  if (config.transformAliasedImportPaths) {
+    beforeCustomTransformers.push(rewriteAliasedSourceFileImportPaths);
+  }
+
+  beforeCustomTransformers.push(
     nativeComponentTransform(compilerCtx, transformOpts),
     proxyCustomElement(compilerCtx, transformOpts),
-    removeCollectionImports(compilerCtx),
-  ];
+    removeCollectionImports(compilerCtx)
+  );
+  return beforeCustomTransformers;
+};
+
+/**
+ * Generate a collection of transformations that are to be applied as a part of the `afterDeclarations` step in the
+ * TypeScript compilation process.
+ *
+ * TypeScript handles the generation of JS and `.d.ts` files through different pipelines. One (possibly surprising)
+ * consequence of this is that if you modify a source file using a transformer, it will not automatically result in
+ * changes to the corresponding `.d.ts` file. Instead, if you want to, for instance, rewrite some import specifiers in
+ * both the source file _and_ its typedef you'll need to run a transformer for both of them.
+ *
+ * See here: https://github.com/itsdouges/typescript-transformer-handbook#transforms
+ * and here: https://github.com/microsoft/TypeScript/pull/23946
+ *
+ * This quirk is not terribly well documented, unfortunately.
+ *
+ * @param config the Stencil configuration associated with the current build
+ * @returns a collection of transformations that should be applied to the source code, intended for the
+ * `afterDeclarations` part of the pipeline
+ */
+const getAfterDeclarationsCustomTransformers = (
```

Comment:
similarly I'm wondering if this should be `getCustomAfterDeclarationsTransformers`

Reply:
I ended up removing these functions in https://github.com/ionic-team/stencil/pull/4501/commits/9b9eb7e3c8df296b28ea8e64b23aeecc1b6c6227

Reply:
it makes sense that you had to put that back into the earlier pre-output target transformation since iirc it's in that run of the typescript compiler that we do `declaration: true` if you have certain options set, so I think adding a transformer for `afterDeclarations` wouldn't have an effect if it's added in the output target code